autocoding/generators.py

```diff
@@ -1,71 +1,140 @@
+                    + util.capitalize(field.attrib["id"])
+                    + "();\n"
+                )
+                headers += (
+                    "void set"
+                    + util.capitalize(struct.attrib["id"])
+                    + util.capitalize(field.attrib["id"])
+                    + "("
+                    + fieldType
+                    + " val);\n\n"
+                )
 
-#                 headers += (
-#                     fieldType
-#                     + " "
-#                     + util.getGetReference(struct, field)
-#                     + "("
-#                     + ("int index" if isArray else "")
-#                     + ");\n"
-#                 )
-#                 headers += (
-#                     "void "
-#                     + util.getSetReference(struct, field)
-#                     + "("
-#                     + fieldType
-#                     + " val"
-#                     + (", int index" if isArray else "")
-#                     + ");\n\n"
-#                 )
-#         out += "} " + struct.attrib["id"] + "_t;\n\n"
-#     return out + "\n" + headers + "\n\n\n"
+                headers += (
+                    fieldType
+                    + " "
+                    + util.getGetReference(struct, field)
+                    + "("
+                    + ("int index" if isArray else "")
+                    + ");\n"
+                )
+                headers += (
+                    "void "
+                    + util.getSetReference(struct, field)
+                    + "("
+                    + fieldType
+                    + " val"
+                    + (", int index" if isArray else "")
+                    + ");\n\n"
+                )
+        out += "} " + struct.attrib["id"] + "_t;\n\n"
+    return out + "\n" + headers + "\n\n\n"
+
+
+def generateTriggersHeader(data):
+    out = ""
+    for field in data.iter("extern"):
+        out += "extern " + field.attrib["type"] + " " + field.attrib["id"] + ";\n"
+    out += "State* eStopTrigger();\n"
+    for field in data.iter("trigger"):
+        out += (
+            "State* state"
+            + field.attrib["currState"]
+            + "To"
+            + field.attrib["nextState"]
+            + "Trigger();\n"
+        )
+    return out
+
+
+def generateTriggers(data):
+    out = ""
+    for extern in data.iter("extern"):
+        out += (
+            extern.attrib["type"]
+            + " "
+            + extern.attrib["id"]
+            + " = "
+            + extern.attrib["initVal"]
+            + ";\n"
+        )
+    out += "State* eStopTrigger() { return hitEStop ? &eStop : 0; }"
+    for trigger in data.iter("trigger"):
+        out += (
+            "State* state"
+            + trigger.attrib["currState"]
+            + "To"
+            + trigger.attrib["nextState"]
+            + "Trigger() "
+        )
+        for cond in trigger:
+            out += (
+                "{ return ("
```

Comment:
I think we should consider the case where there are more than 2 conditions that trigger a true return

Reply:
I created the template files while testing. I could push them so you could look over them. Triggers.cpp and Triggers.h are based on the same xml file. I couldn't figure out how to use the new xml file with autocode so I moved the xml code for Triggers to data.xml and I'll look over the comments on the code by tomorrow.